README.md

```diff
@@ -1,32 +1,127 @@
-# Gradle Lifecycle
+# Lifecycles, phases, and goals
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Define the Gradle Lifecycle
+- Define the relationships between Lifecycle Phases, Tasks, and Plugins.
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+A Gradle build follows a clearly defined process to deploy and distribute a project.  
 
-## Topic 1
+- A **lifecycle** is defined by a list of phases. 
+- A **phase** represents a stage of a lifecycle.
+- A **task** is a piece of work that a build performs.
+- A **plugin** is an artifact that provides tasks.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+## Lifecycle Phases
 
-## Topic 2
+Every Gradle build proceeds through three lifecycle phases
+in precisely the same order: initialization, configuration, and execution.
+
+- **Initialization**:  Gradle locates the build files it must process and
+  determines whether the build is single-project or multi-project. The build files
+  are passed to the next phase.
+- **Configuration**: Gradle executes each build file as a Groovy or Kotlin script
+  to create a directed acyclic graph (DAG) of task objects.
+  - “Directed” means each dependency arrow goes in one direction. 
+  - “Acyclic” means that there are no loops in the graph.
+- **Execution**: Gradle identifies the tasks in the task DAG  
+  and executes them in dependency order to ensure each task is executed only once.
+
+For example, the following image shows the DAG for a Java build:
+
+![java plugin DAG](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/java-dag.png)
+
+Each task defines a set of inputs and outputs.
+The inputs for **compileJava** are the Java source files, while the
+outputs are class files. If none of the inputs or outputs
+have changed, Gradle will skip that task. 
+We call this Gradle’s **incremental build support**.
+
+Let's explore some of this using IntelliJ's Gradle Tool Window,
+which can be opened either by clicking on "Gradle" along
+the edge of the IntelliJ window, or by selecting ** View | Tool Windows | Gradle**
+from the main menu.
+  
+![gradle tool window](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/gradletoolwindow.png)
+
+### Tasks
+
+Expand the **Tasks** folders to see tasks available for execution.
+
+![gradle tasks](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/gradletasks.png)
+
+- Gradle's Base plugin provides general lifecycle tasks such as *clean*,
+  *assemble*, and *build*.
+- The Java plugin provides additional tasks related to compiling and
+  packaging a Java project: *compileJava*, *processResources*, *classes*,
+  *compileTestJava*, *processTestResources*, *testClasses*, *jar*, *javadoc*,
+  *test*, and *cleanTaskName*.
+
+We can get a description of each task by typing "gradle tasks --all"
```

Comment:
Possibly do backticks here instead of double quotes?

`gradle tasks --all`

Reply:
fixed